002 dev

VladislavKolos opened the pull request:
002 feat: add migrations, DTOs, mappers, services, and controllers
- added database migrations for initial data population
- created DTOs for request/response handling
- implemented entity mappers
- updated and extended repositories
- introduced global constants and URI enum
- developed core services
- built REST controllers 


002 feat: extended repositories, refined data model, implemented core logic and endpoints, added utility classes

VladislavKolos committed:
001 fix after review

VladislavKolos committed:
Merge branch 'refs/heads/001-dev' into 002-dev

# Conflicts:
#	src/main/java/org/example/api/hotel/repository/HotelRepository.java

VladislavKolos committed:
002 feat: add migrations, DTOs, mappers, services, and controllers

- added database migrations for initial data population
- created DTOs for request/response handling
- implemented entity mappers
- updated and extended repositories
- introduced global constants and URI enum
- developed core services
- built REST controllers

VladislavKolos committed:
002 feat: extended repositories, refined data model, implemented core logic and endpoints, added utility classes

VladislavKolos committed:
002 feat: add validation (custom annotations & validator classes), implement exception handling, and refactor code

Caromeen commented on src/main/java/org/example/api/hotel/controller/HotelAmenityController.java:
```diff
@@ -0,0 +1,44 @@
+        log.info("Amenities added to hotel with ID: {} successfully", response.id());
+
+        String resourceUri = UriUtil.createUri(UriType.HOTEL_AMENITY, response.id());
+        log.info("Hotel with amenities resource URI created: {}", resourceUri);
```
контролллер логируует респонс/реквест
остальные логи непосредственно в Сервисе, не контроллер создал uri/ добавил отель, он ток принял req/отдал response

Caromeen replied:
check logbook

VladislavKolos committed:
002 feat: implement quick profile & DB switching
    fix: resolve various bugs
    refactor: improve code structure and readability
    test: add unit and integration tests
    docker: set up test DB with Docker
    docs: create Swagger docs and README.md
